inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -22,203 +22,37 @@
-                                        volume=form_fields.get('series_volume')
-                                        )
-            builder.add_book(
-                publisher=form_fields.get('publisher_name'),
-                place=form_fields.get('publication_place'),
-                date=form_fields.get('publication_date'))
-
-    builder.add_publication_info(
-        year=year,
-        cnum=form_fields.get('conference_id'),
-        journal_issue=form_fields.get('issue'),
-        journal_title=form_fields.get('journal_title'),
-        journal_volume=form_fields.get('volume'),
-        page_start=form_fields.get('start_page'),
-        page_end=form_fields.get('end_page'),
-        artid=form_fields.get('artid'),
-        parent_record=form_fields.get('parent_book')
-    )
-
-    builder.add_accelerator_experiments_legacy_name(
-        legacy_name=form_fields.get('experiment')
-    )
-
-    language = form_fields.get('other_language') \
-        if form_fields.get('language') == 'oth' \
-        else form_fields.get('language')
-    builder.add_language(language=language)
-
-    if form_fields.get('title_translation'):
-        builder.add_title_translation(
-            title=form_fields['title_translation'],
-            language='en',
-        )
-
-    builder.add_title(
-        title=form_fields.get('title_arXiv'),
-        source='arXiv'
-    )
-
-    builder.add_title(
-        title=form_fields.get('title_crossref'),
-        source='crossref'
-    )
-
-    builder.add_license(url=form_fields.get('license_url'))
-
-    builder.add_public_note(public_note=form_fields.get('public_notes'))
-
-    builder.add_public_note(
-        public_note=form_fields.get('note'),
-        source='arXiv' if form_fields.get('categories') else 'CrossRef'
-    )
-
-    form_url = form_fields.get('url')
-    form_additional_url = form_fields.get('additional_url')
-    if form_url and not _is_arxiv_url(form_url):
-        obj.extra_data['submission_pdf'] = form_url
-        if not form_additional_url:
-            builder.add_url(url=form_url)
-
-    if form_additional_url and not _is_arxiv_url(form_additional_url):
-        builder.add_url(url=form_additional_url)
-
-    [builder.add_report_number(
-        report_number=report_number.get('report_number')
-    ) for report_number in form_fields.get('report_numbers', [])]
-
-    builder.add_collaboration(collaboration=form_fields.get('collaboration'))
-
-    builder.add_acquisition_source(
-        datetime=datetime.datetime.utcnow().isoformat(),
-        submission_number=obj.id,
-        internal_uid=int(obj.id_user),
-        email=form_fields.get('email'),
-        orcid=form_fields.get('orcid'),
-        method='submitter'
+def _get_user_comment(workflow_object):
+    user_comment = next(
+        (
+            note.get('value')
+            for note in workflow_object.data.get('_private_notes', [])
+            if note.get('source') == 'submitter'
+        ),
+        '',
```

Comment:
You already defined this in inspirehep/modules/authors/tasks.py

Reply:
I think that when I started the PR they were different... looking to see if I messed up a copy-paste or something.

Reply:
yes, this looks like a messed up copy-paste.

And the goal here is to do the `formdata_to_model` only once at the start of the workflow, then save the results in the `source_data` so when we restart the workflow, there will be no need to rerun the `fromdata_to_model` again, decoupling the workflow from the form, and allowing a generic handling of them.
Otherwise the source truth for the workflow will not always be the `data` and `extra_data` under `source_data` but sometimes they would be generated by the other `formdata`.

Reply:
@ammirate So if I fix it, I end up with the same method in both places... but I'm not sure if it's worth putting it anywhere else.

It's very specific, so I'm reluctant on putting it in inspire-utils, adding it to inspirehep/modules/tools does not look right either :/, I would not put it in inspirehep/modules/__init__.py either.

Maybe under inspirehep/modules/forms/utils.py? wdyt?

Reply:
hmpf... it looks to me also out of context there :/